src/rpc/connection/crypto.c

```diff
@@ -446,6 +454,16 @@ int crypto_recv_initiate(struct crypto_context *cc, unsigned char *data)
 
   cc->state = TUNNEL_ESTABLISHED;
 
+  char *pluginkey_base16_encoded = MALLOC_ARRAY((PLUGINKEY_SIZE*2) + 1, char);
```

Comment:
why use heap memory? i don't like the malloc. a stack variable would be nicer:

`char plugin_base16_encoded[(PLUGINKEY_SIZE*2) + 1]`